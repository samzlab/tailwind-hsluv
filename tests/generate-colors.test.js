const { generateColors, resolveColor } = require('../');

describe('Generate colors', () => {

	test('"red" should resolve to "#ff0000"', () => {
		expect(resolveColor('red')).toEqual({ hex: 'FF0000', rgb: [ 255, 0, 0 ] });
	});

	test('color variations has default key', () => {
		expect(generateColors({
			red: 'red'
		})).toHaveProperty([ 'red', 'default' ]);
	});

	test('variations (step=50) contains 50...950', () => {
		const colors = generateColors({ red: 'red' }, { step: 50 });

		expect(colors).toHaveProperty([ 'red', '50' ]);
		expect(colors).toHaveProperty([ 'red', '950' ]);
	});

	test('invalid step values (510, 45) should throw Error', () => {
		const colorMap = { red: 'red' };

		expect(() => generateColors(colorMap, { step: 510 })).toThrow(Error);
		expect(() => generateColors(colorMap, { step: 45 })).toThrow(Error);
	});

	test('by name "red"', () => {
		expect(generateColors({ red: 'red' })).toMatchSnapshot();
	});

});