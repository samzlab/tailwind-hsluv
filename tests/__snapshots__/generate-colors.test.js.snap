// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate colors by name "red" 1`] = `
Object {
  "red": Object {
    "100": "#FFDADA",
    "200": "#FFB3B3",
    "300": "#FF8888",
    "400": "#FF5353",
    "500": "#EF0000",
    "600": "#C00000",
    "700": "#930000",
    "800": "#690000",
    "900": "#410000",
    "default": "#FF0000",
  },
}
`;
